statsmodels/tsa/ar_model.py

```diff
@@ -388,7 +388,15 @@ def _stackX(self, k_ar, trend):
         X = lagmat(endog, maxlag=k_ar, trim='both')
         k_trend = util.get_trendorder(trend)
         if k_trend:
-            X = add_trend(X, prepend=True, trend=trend)
+            try:
+                X = add_trend(X, prepend=True, trend=trend,
+                              has_constant="raise")
+            except ValueError as err:
```

Comment:
instead of catching and reraising to add to the exception message, we could just add an extra message as a keyword options.

only `"Try specifying trend='nc'"` is not generic, and the message can be issued by add_trend

Reply:
I think this is not correct. The data doesn't actually have a constant column, it is just locally constant, and add_trend is doing the wrong thing.  IMO the correct thing to do is to always add exactly the same trend that was in the model, so `has_constant='add'`

Reply:
> just add an extra message as a keyword options.

Please no.  Expanding the API of add_trend for a single use just adds complexity to that function with nearly no gain.  IMO @jbrockmendel 's solution is the most standard Pythonic way to swap an error message (although I think the approach is not correct)

Reply:
There are many standard Pythonic ways that we avoid.
Reraising an exception just to add a few words to the exception message would be too much of an extra layer IMO.
We have control over both functions, while the "standard Pythonic way" is useful if we don't have control over the raising function.

(By the same idea we could catch a few hundred numpy, pandas and scipy exceptions and add more context information.)

Reply:
> The data doesn't actually have a constant column, it is just locally constant

Fair enough; notwithstanding the other issues, the message could be made more accurate.

> Please no. Expanding the API of add_trend for a single use just adds complexity to that function with nearly no gain

Agreed.

> IMO the correct thing to do is to always add exactly the same trend that was in the model, so has_constant='add'

Just tried this.  I expected the `fit` call to raise because the RHS had collinear regressors, but it looks like this inherits `OLS`'s behavior and allows this.  So the `fit` call goes through, but we end up with an `ARResults` object with `bse` that's all-`NaN`

If I tilt my head and squint a bit this behavior kind of makes sense, but I think raising would be more reasonable.

Two other problems with raising the way this PR currently does:
 - it could raise from within `select_order`, where the correct behavior would be to recover gracefully and go to a smaller order.
 - Suggesting `trend="nc"` makes sense of the user passed `trend="c"`, but not if the user passed `trend="ct"`